src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -36,53 +49,75 @@ const useGroupSetting = ({navigation, route}) => {
-        changeImageUrl = res.data.url;
-      } catch (e) {
-        setIsLoading(false);
-        return;
-      }
+  React.useEffect(() => {
+    if (selectedChannel) {
+      setUrlImage(selectedChannel?.channelPicture);
     }
+  }, [selectedChannel]);
 
-    if (changeName || changeImage) {
-      if (withLoading) setIsLoading(true);
-      let dataEdit = {
-        name: groupName
-        // ...(changeImage && {image: base64Profile}),
+  const submitData = async (withNavigation = true) => {
+    if (!changeImage && !changeName) return navigation.goBack();
+    try {
+      setLoadingUpdate(true);
+      let body = {
+        channel_id: route?.params?.id,
+        channel_name: groupName,
+        channel_image: urlImage
       };
-      if (changeName) {
-        dataEdit = {...dataEdit, isEditName: true};
+      if (urlImage) {
+        try {
+          const res = await uploadPhotoImage(urlImage);
+          if (res) {
+            body = {...body, channel_image: res.data.url};
+          }
+        } catch (e) {
+          console.log(e, 'error upload');
+        }
       }
-      if (changeImage) {
-        dataEdit.image = changeImageUrl;
-      } else if (channel?.data?.image) {
-        dataEdit.image = channel?.data?.image;
+      const response = await SignedMessageRepo.editChannel(body);
+      let newChannel = response.data?.channel;
+      newChannel = {
+        ...chat.selectedChannel,
+        ...newChannel,
+        channelType: handleChannelType(newChannel?.channelType),
+        channelPicture: body?.channel_image
+      };
+      ChannelList.updateChannelInfo(
+        localDb,
+        body.channel_id,
+        body.channel_name,
+        body.channel_image
+      );
+      if (newChannel) {
+        setChat({
+          ...chat,
+          selectedChannel: newChannel
+        });
       }
-      console.log({dataEdit}, 'nusu');
-      updateDataEdit(dataEdit, withNavigation);
-    } else if (withNavigation) navigation.goBack();
-  };
+      refresh('channelInfo');
+      refresh('chat');
 
-  const updateDataEdit = async (dataEdit, withNavigation) => {
-    try {
-      await channel.update(dataEdit);
-      if (withNavigation) navigation.navigate('ChatDetailPage');
-    } catch (e) {
-      if (__DEV__) {
-        console.log(`error : ${e}`);
+      refresh('channelList');
+
+      if (withNavigation) {
+        navigation.goBack();
       }
+      setLoadingUpdate(false);
+    } catch (e) {
+      setLoadingUpdate(false);
       SimpleToast.show(StringConstant.groupSettingUpdateFailed);
     }
-    setIsLoading(false);
+  };
+
+  const handleChannelType = (channelType) => {
+    if (channelType === 1) {
+      return CHANNEL_GROUP;
+    }
+    return channelType;
   };
 
   const lounchGalery = async () => {
-    const {success, message} = await requestExternalStoragePermission();
+    const {success} = await requestExternalStoragePermission();
     if (success) {
       launchImageLibrary(
         {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [119-134]

In the `lounchGalery` function, when the permission request fails, the app redirects the user to the settings page without any warning or explanation. It would be better to inform the user about why they are being redirected to the settings page. Consider adding a toast message or alert before redirecting.

```diff
    } else {
+      SimpleToast.show("Permission denied. Please enable storage access from settings.");
      Linking.openSettings();
     }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->